pkg/jobmon/spool.go

```diff
@@ -0,0 +1,256 @@
+package jobmon
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/nightlyone/lockfile"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/cloudradar-monitoring/cagent/pkg/common"
+)
+
+const (
+	markerRunning         = "0"
+	markerFinished        = "1"
+	spoolDirPermissions   = 6777
+	spoolEntryPermissions = 0666
+	jsonExtension         = "json"
+)
+
+var ErrJobAlreadyRunning = errors.New("A job with same ID is already running")
+
+type SpoolManager struct {
+	dirPath string
+	lock    *lockfile.Lockfile
+	logger  *logrus.Logger
+}
+
+// NewSpoolManager creates a new object to manage jobmon spool
+// dirPath must be absolute path
+func NewSpoolManager(dirPath string, logger *logrus.Logger) *SpoolManager {
+	lockFile, _ := lockfile.New(fmt.Sprintf("%s/spool.lock", dirPath))
```

Comment:
in case the directory is missing, we will get nil dereference on line 204

Reply:
lockfile is an external library.  `New()` call just constructs the object. Filesystem is not accessed

Reply:
ah, I see. It just a wrapped string